src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,443 @@
+        address initialOwner,
+        IPauserRegistry _pauserRegistry,
+        uint256 initialPausedStatus
+    ) external initializer {
+        _initializePauser(_pauserRegistry, initialPausedStatus);
+        _transferOwnership(initialOwner);
+    }
+
+    /**
+     * @notice Called by the delagation manager to set delay when operators register.
+     * @param operator The operator to set the delay on behalf of.
+     * @param delay The allocation delay in seconds.
+     * @dev msg.sender is assumed to be the delegation manager.
+     */
+    function setAllocationDelay(address operator, uint32 delay) external {
+        require(msg.sender == address(delegation), OnlyDelegationManager());
+        _setAllocationDelay(operator, delay);
+    }
+
+    /**
+     * @notice Called by operators to set their allocation delay.
+     * @param delay the allocation delay in seconds
+     * @dev msg.sender is assumed to be the operator
+     */
+    function setAllocationDelay(
+        uint32 delay
+    ) external {
+        require(delegation.isOperator(msg.sender), OperatorNotRegistered());
+        _setAllocationDelay(msg.sender, delay);
+    }
+
+    /**
+     * @notice This function takes a list of strategies and adds all completable modifications for each strategy, 
+     * updating the encumberedMagnitude of the operator as needed.
+     *
+     * @param operator address to complete modifications for
+     * @param strategies a list of strategies to complete modifications for
+     * @param numToClear a list of number of pending modifications to complete for each strategy
+     *
+     * @dev can be called permissionlessly by anyone
+     */
+    function clearModificationQueue(
+        address operator,
+        IStrategy[] calldata strategies,
+        uint16[] calldata numToClear
+    ) external onlyWhenNotPaused(PAUSED_MODIFY_ALLOCATIONS) {
+        require(strategies.length == numToClear.length, InputArrayLengthMismatch());
+        require(delegation.isOperator(operator), OperatorNotRegistered());
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            _clearModificationQueue({operator: operator, strategy: strategies[i], numToClear: numToClear[i]});
+        }
+    }
+
+    /**
+     * @notice Modifies the propotions of slashable stake allocated to a list of operatorSets for a set of strategies
+     * @param allocations array of magnitude adjustments for multiple strategies and corresponding operator sets
+     * @dev Updates encumberedMagnitude for the updated strategies
+     * @dev msg.sender is used as operator
+     * @dev For each allocation, allocation.operatorSets MUST be ordered in ascending order according to the
+     * encoding of the operatorSet. This is to prevent duplicate operatorSets being passed in. The easiest way to ensure
+     * ordering is to sort allocated operatorSets by address first, and then sort for each avs by ascending operatorSetIds.
+     */
+    function modifyAllocations(
```

Comment:
Should we include a signature based variant, so anyone can modify allocations on behalf of an operator?

Reply:
removed because it was bloating code and we are adding delegated accounts

wdyt @ypatil12 @wadealexc ?

Reply:
leave to delegated accounts IMO